src/phoenix/experimental/evals/models/openai.py

```diff
@@ -287,3 +292,17 @@ def get_tokens_from_text(self, text: str) -> List[int]:
 
     def get_text_from_tokens(self, tokens: List[int]) -> str:
         return self.encoder.decode(tokens)
+
+    @property
+    def supports_function_calling(self) -> bool:
```

Comment:
Needs to be on the base class.

Reply:
I was ambivalent because I don't know what function calling looks like for other models, i.e. if there is another (non-OpenAI) model for which this property is True, it's not guaranteed for us to be able to transfer the calling procedure from OpenAI to that model directly. Since how we use this property is strictly predicated on the model being OpenAI, I am hesitant for the time being to overload the base class with an additional property that is pretty much useless for all other models.

Reply:
I see, you're doing a type check to make sure the model is an instance of `OpenAIModel` before using this attribute. Makes sense.